test/metabase/query_processor_test/nested_queries_test.clj

```diff
@@ -350,6 +350,43 @@
               (run-native-query (str native-sub-query " -- small comment here\n")))
           "Ensure trailing comments followed by a newline are trimmed and don't cause a wrapping SQL query to fail"))))
 
+(deftest ^:parallel card-id-native-source-query-with-long-alias-test
```

Comment:
Absolute carnage in drivers tests. Maybe this needs to be backend specific? Was hoping to avoid that. Summary of test failures below. I feel like most of these are fixable, but maybe I need to specifically exclude bigquery and sqlserver?

- mariadb/mysql/postgres/sparksql/sqlite
  - qp result cols differ. case issue? maybe need regex match instead...
  - :display_name "Total Number Of People From Each State Separated By 165e2c32"
  - :display_name "Total Number Of People From Each State Separated By 00028d48"
  - :name "TOTAL_NUMBER_OF_PEOPLE_FROM_EACH_STATE_SEPARATED_BY_165e2c32"
  - :name "Total_number_of_people_from_each_state_separated_by_00028d48"
- athena/oracle/redshift/snowflake/vertica
  - claims "people" table does not exist. case issue?
  - TABLE_NOT_FOUND: line 2:125: Table 'awsdatacatalog.default.people' does not exist
  - Error executing query: ORA-00942: table or view does not exist
  - Error executing query: ERROR: relation "people" does not exist
  - Object 'PEOPLE' does not exist or not authorized.
  - ERROR: Relation "people" does not exist
- bigquery
  - Table "people" must be qualified with a dataset (e.g. dataset.table)
- presto
  - not sure failed to print exception
- sqlserver
  - rejects the long column name
  - Invalid column name 'Total_number_of_people_from_each_state_separated_by_state_and_then_we_do_a_count'

Reply:
Turns out hard coding SQL for a test that's supposed to be driver agnostic is a bad idea. I updated the test to generate the inner native query via `qp.compile/compile` instead.

The remaining failures here on sqlserver and presto are related to the fact that the inner native query contains an `ORDER BY` clause due to the `:breakout`, and neither driver like it. sqlserver rejects the query as invalid, and presto just silently ignores the `ORDER BY` and returns unordered results. I need to check if this is a bug or actually invalid SQL as sqlserver claims. Not sure what do here, I could either: (A) try to strip off the `ORDER BY` clause from the compiled inner query (gross), (B) skip these drivers for this test, (C) maybe prevent `qp.compile` from adding the `ORDER BY` clause in the first place and instead tack it on to the outer query?

Reply:
Fixed by moving the `LIMIT` clause into the inner query.